### PR TITLE
Finish backup recipes from soakbot

### PR DESCRIPTION
**Pull request by DarkmatterVale**
I apologize this PR is so large.

In this PR I have added recipes for gnurobbo 0.66, transplus, drive_encryption (which is not working), dovecot, and sockhop. This PR should allow all pr_backup branches at https://github.com/soakbot/haikuports to be closed.

### CODE REVIEW

File: games-puzzle/gnurobbo/gnurobbo-0.66.recipe [hunk 1]
```diff
@@ -0,0 +1,50 @@
+SUMMARY="GNU Robbo: Classic Atari Puzzle Game"
+DESCRIPTION="An open source reimplementation \
+of Janusz Pelc's Robbo for the Atari XE/XL which was distributed \
+by LK Avalon in 1989. In this retro game, you can collect and reach the \
+UFO-like object."
+HOMEPAGE="http://gnurobbo.sourceforge.net/"
+COPYRIGHT="2002-2010 The GNU Robbo Team"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+CHECKSUM_SHA256="54d2418f49e52b8f49631eb93af7789074b628d308a6d26b00cc795ad0b04d89"
```
Comment (humdingerb) [inline games-puzzle/gnurobbo/gnurobbo-0.66.recipe]: CHECKSUM_SHA256 under SOURCE_URI

File: haiku-apps/drive_encryption/drive_encryption-1.0.recipe [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+DESCRIPTION="DriveEncryption is a suite of programs to allow you to \
+encrypt devices and file images. The encryption is done via the AES algorithm, \
+and protected via a password.
+It can also use devices created with TrueCrypt (http://www.truecrypt.org/), \
+available for Windows and Linux, as long as you've created them to use the \
+AES encryption algorithm and a RIPEMD-160 hash algorithm protected password."
+HOMEPAGE="https://github.com/chirayudesai/driveencryption"
+COPYRIGHT="2007-2014 pinc Software"
+LICENSE="
+	TrueCrypt
```
Comment (humdingerb) [inline haiku-apps/drive_encryption/drive_encryption-1.0.recipe]: TrueCrypt in the same line as LICENSES.

File: haiku-apps/drive_encryption/drive_encryption-1.0.recipe [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	MIT
+	"
+REVISION="1"
+CHECKSUM_SHA256=""
```
Comment (humdingerb) [inline haiku-apps/drive_encryption/drive_encryption-1.0.recipe]: CHECKSUM_SHA256 under SOURCE_URI. CHECKSUM_SHA256 shouldn't be emtpy... (?)

Comment (DarkmatterVale) [inline haiku-apps/drive_encryption/drive_encryption-1.0.recipe]: I can't get a CHECKSUM for a commit that doesn't exist yet. As is stated in the commit, I need work from axel on the actual app before the recipe will be finished. I can move it below SOURCE_URI though.

File: haiku-apps/drive_encryption/drive_encryption-1.0.recipe [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		pushd $i
+		make $jobArgs OBJ_DIR=objects
+		popd
+	done
+}
+
+INSTALL()
+{
+	mkdir -p $addOnsDir/kernel/drivers/bin $addOnsDir/kernel/drivers/dev/disk
+        cp -a driver/objects/encrypted_drive $addOnsDir/kernel/drivers/bin
```
Comment (humdingerb) [inline haiku-apps/drive_encryption/drive_encryption-1.0.recipe]: there seems to be some whitespace issue in the cp and ln line.

File: haiku-apps/drive_encryption/drive_encryption-1.0.recipe [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+and protected via a password.
+It can also use devices created with TrueCrypt (http://www.truecrypt.org/), \
+available for Windows and Linux, as long as you've created them to use the \
+AES encryption algorithm and a RIPEMD-160 hash algorithm protected password."
+HOMEPAGE="https://github.com/chirayudesai/driveencryption"
+COPYRIGHT="2007-2014 pinc Software"
+LICENSE="
+	TrueCrypt
+	MIT
+	"
```
Comment (humdingerb) [inline haiku-apps/drive_encryption/drive_encryption-1.0.recipe]: Closing " in the "MIT" line

File: haiku-apps/drive_encryption/drive_encryption-1.0.recipe (hunk 4 above)
Comment (DarkmatterVale) [inline haiku-apps/drive_encryption/drive_encryption-1.0.recipe]: Sorry I missed that, turns out there was way too many spaces (not even using tabs). It should be fixed now.

File: haiku-apps/drive_encryption/drive_encryption-1.0.recipe [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+SOURCE_DIR="driveencryption-993a11ceefac15d77406b697afb60b177d99bc45"
+
+ARCHITECTURES="x86_gcc2 ?x86 ?x86_64"
+
+PROVIDES="
+	drive_encryption = $portVersion
+	cmd:encrypted_drive_control = $portVersion
+	app:DriveEncryption = $portVersion
+	app:Login = $portVersion
+	"
```
Comment (humdingerb) [inline haiku-apps/drive_encryption/drive_encryption-1.0.recipe]: the add-on is missing: "add_on:encrypted_drive"

File: games-puzzle/gnurobbo/gnurobbo-0.66.recipe [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	"
+
+BUILD()
+{
+	make $jobArgs PACKAGE_DATA_DIR=$appsDir/gnurobbo/data
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir/gnurobbo
```
Comment (diversys) [inline games-puzzle/gnurobbo/gnurobbo-0.66.recipe]: $appsDir/GNURobbo

**Pull request merged**